### PR TITLE
#389  Save media

### LINKED ISSUE #389: Save image files from test_db. stb2
Now files no saving leads us to 404 errors in tests.
See #359 to find how errors look like

### PR DESCRIPTION
**Pull request by duker33**
Closes #389 

stb2 after this PR

### CODE REVIEW

File: shopelectro/management/commands/test_db.py [hunk 1]
```diff
@@ -9,13 +9,13 @@
 - now you have json file, that'll be used by our TDD tests
 """
 import os
-
+import shutil
 from django.conf import settings
```
Comment (artemiy312) [inline shopelectro/management/commands/test_db.py]: Please, add a line before third-party packages imports

File: shopelectro/management/commands/test_db.py [hunk 2]
```diff
@@ -118,11 +121,20 @@ def get_name(number, parent=None):
...
             def create_image(file_path, slug):
-                Image.objects.create(
+                # product "/catalog/products/2/" contains image
+                image = Image.objects.create(
                     model=page,
                     slug=slug,
                     image=ImageFile(open(file_path, mode='rb'))
                 )
+                # save files to media folder
+                file_src = open(file_path, mode='rb')
```
Comment (artemiy312) [inline shopelectro/management/commands/test_db.py]: Open a file with `with-statement` to release it after use
```python
with open(file_path, mode='rb') as file:
    image = Image.objects.create(
        model=page,
        slug=slug,
        image=ImageFile(file)
    )
    ...
```

File: shopelectro/management/commands/test_db.py [hunk 3]
```diff
@@ -118,11 +121,20 @@ def get_name(number, parent=None):
...
+                file_name = os.path.split(file_src.name)[-1]
```
Comment (artemiy312) [inline shopelectro/management/commands/test_db.py]: Let's use [os.path.basepath](https://docs.python.org/3/library/os.path.html#os.path.basename) instead

File: shopelectro/management/commands/test_db.py (hunk 1 above)
Comment (duker33) [reply shopelectro/management/commands/test_db.py]: TODO

File: shopelectro/management/commands/test_db.py (hunk 2 above)
Comment (duker33) [reply shopelectro/management/commands/test_db.py]: TODO

File: shopelectro/management/commands/test_db.py (hunk 3 above)
Comment (duker33) [reply shopelectro/management/commands/test_db.py]: TODO

File: shopelectro/management/commands/test_db.py (hunk 1 above)
Comment (artemiy312) [reply shopelectro/management/commands/test_db.py]: @duker33 the comment above is still not resolved

File: shopelectro/management/commands/test_db.py [hunk 4]
```diff
@@ -118,11 +121,20 @@ def get_name(number, parent=None):
     def create_products(self, categories, tags):
         def create_images(page: Page):
             def create_image(file_path, slug):
-                Image.objects.create(
+                # product "/catalog/products/2/" contains image
+                image = Image.objects.create(
                     model=page,
                     slug=slug,
                     image=ImageFile(open(file_path, mode='rb'))
```
Comment (artemiy312) [inline shopelectro/management/commands/test_db.py]: We are opening here the same file as below and does not release it. We should group these lines by one file's context:
```python
with open(file_path, mode='rb') as file_:
    image = Image.objects.create(
        ...,
        image=ImageFile(file_)
    )
    ...
    shutil.copyfile(file_.name, file_dst_path)
```

Comment (duker33) [reply shopelectro/management/commands/test_db.py]: Done.

**Comment by duker33** [comment]
created separated task https://github.com/fidals/stroyprombeton/issues/237 for stb

**Pull request merged**